File: 7.reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        negative = (x < 0)
        
        x = int(str(abs(x))[::-1])

        if negative:
            x *= -1
        
        min = -2**31
        max = 2**31 - 1

        if min <= x <= max:
            return x

        return 0

File: 7.reverse-integer.rs
impl Solution {
    pub fn reverse(x: i32) -> i32 {
        let num = x.abs().to_string().chars().rev().collect::<String>().parse::<i32>();
        if let Ok(res) = num {
            res * x.signum()
        } else {
            0
        }
    }
}
